addons/product/models/product_template.py

```diff
@@ -602,6 +602,15 @@ def _create_variant_ids(self):
                 variants_to_create += current_variants_to_create
                 variants_to_activate += current_variants_to_activate
 
+            else:
+                ptav_ids = lines_without_no_variants.product_template_value_ids.ids
+                active_existing_variants = {variant.product_template_attribute_value_ids: variant for variant in all_variants if variant.active}
+                active_variants_attribute_line_ids = list(set(itertools.chain(*[v.attribute_line_id.ids for v in active_existing_variants])))
+                if active_variants_attribute_line_ids == lines_without_no_variants.ids:
+                    for attribute_values in active_existing_variants:
+                        if attribute_values and all([av_id in ptav_ids for av_id in attribute_values.ids]):
+                            current_variants_to_activate += active_existing_variants[attribute_values]
```

Comment:
All in all I think the solution is to iterate on `all_variants` and add them to `current_variants_to_activate` based on the result of `_is_variant_possible`.

This is the simple case, but on top of that we need to ensure that 2 variants with the same combination are not activated (it can be possible in some situations that there are multiple archived variants for the same combination, and we must only activate one of them). Add a test for this case too.

Reply:
I don't understand the second part, there is a unique constraint product_product_combination_unique which does not allow to have the same combination twice. How can we reproduce a such use case?

Reply:
The unique rule is only for active products. You can manually generate a second archived variant for the same combination.

There used to be flows that lead to have multiple variants for the same combination, but I think they have been fixed (the last one was probably the one you are fixing with this PR). It's still possible coming from migrated databases to be in this case.